Fix logic that determines whether to spoof GetModuleFileNameW(nullptr)

rfortier opened the pull request:
Update: TiltedCore v0.2.8 release has now been cut, so updated with that version.
PR, although complete barring more comments, will remain draft until prerequisites met:

1. xmake_repo accepts the PR for the new version of TiltedCore.
2. PRs to update other modules can then be submitted. TiltedHooks, TiltedReverse, and TiltedUI need a version pin update.
3. Those PRs merged. 
4. Then, finally, this PR can be moved converted from Draft to Submitted and merged.
---
It was depending on figuring out where it was called from, but the logic was unreliable and there is no apparent way to fix it. Multiple attempts using more sophisticated FOSS stack crawlers all failed.

Fixes: now, it switches spoofing on just before loading SkyrimSE.exe (so the loaded DLLs see the path they expect).

The required companion fix in TIltedCore:TiltedPhoques::GetPath() persists the value found originally, so it doesn't change out from under with late callers. If your code has only late callers, just add a call before loading your game binary.

rfortier committed:
Fix logic that determines whether to spoof GetModuleFileNameW(nullptr)

It was depending on figuring out where it was called from, but the logic was unreliable and there is no apparent way to fix it. Now, it switches spoofing on just before loading SkyrimSE.exe (so the loaded DLLs see the path they expect).

Requires a companion fix in TIltedCore:TiltedPhoques::GetPath() to persist the value found originally, so it doesn't change out from under with late callers. If you code has only late callers, just add a call before loading your game binary.

Make isLoaded member private (Review comment)

Fix xmake.lua to use TiltedCore 0.2.8

miredirex commented on Code/immersive_launcher/Launcher.h:
```diff
@@ -21,7 +21,11 @@ struct LaunchContext
+    bool isLoaded{false};
     ExeLoader::TEntryPoint gameMain = nullptr;
+
+    void SetLoaded() { isLoaded = true; }   // If loaded, need to spoof GetModuleFileName*(nullptr)
+    bool GetLoaded();
```
Unnecessary getter and setter. The field is public

rfortier replied:
fixed

miredirex commented on Code/immersive_launcher/Launcher.cpp:
```diff
@@ -126,6 +131,7 @@ bool LoadProgram(LaunchContext& LC)
     LC.Version = QueryFileVersion(LC.exePath.c_str());
     if (LC.Version.empty())
         DIE_NOW(L"Failed to query game version");
+    LC.SetLoaded();
```
Why not after the `if (!loader.Load(reinterpret_cast<uint8_t*>(content.data())))` call?

rfortier replied:
Slightly subtle. The goal is for the custom loader to load SkyrimSE.exe with spoofing enabled so any DLLs, in particular any preload DLLs like SSE EngineFixes.dll, experience what looks like SkyrimSE.exe is loading rather than experiencing SkyrimTogether.exe as loaded. 

In particular, attempts to evaluate the directory of the running executable have to return the GamePath, not the  SkyrimTogether.exe path.

So the switch must be thrown before loading the executable.

rfortier committed:
Updated all submodules to latest version.

The new version removes the TiltedCore version pins in the submodules. Submodules inherit version pins from their parents. Not pinning a version means they will all consistently decide to use the version the parent selects, without needing to edit the submodules.

This avoids a crash from TiltedEvolution moving to a new TiltedCore with  the GetPath() fix; if other submodules are on a downrev TiltedCore, a crash is likely.

rfortier commented:
This PR is now ready to merge. 

Updated all submodules to latest commit, which removes the version pin on TiltedCore. This means submodules will consistently use what the parent TiltedEvolution uses, which is 0.2.8. If the submodules load an older version it conflicts and will crash.

Rebased to latest /dev to ensure no merge conflicts.

Squashed some commits.

Left the 4 submodule updates as a separate commit for visibility.

rfortier commented:
Sigh. Back to Draft. 

The latest TiltedUI, before the pin change, is incompatible with the latest TiltedEvolution; TiltedUI expects the keybinds PR to be merged, which it is not. So, F2 stopped working.

rfortier committed:
Review comment. Updating TiltedHooks to latest got ToeKnee's rebind keys PR, which no longer initialized the UI toggle keys in TiltedHooks. So we have to initialize in TiltedEvolution.

rfortier commented:
Review comment bugfix. Moving to _latest_ TiltedHooks got the version created for key rebinds, which no longer initializes the UI toggle keys. Have to initialize them if you don't have PR #724  changes
